nvflare/edge/aggregators/model_update_dxo.py

```diff
@@ -34,45 +33,34 @@ def _aggregate(self, weight_base, weight_to_add):
                 sub_to_add = weight_to_add.get(key)
                 self._aggregate(sub_object, sub_to_add)
             weight_base[key] = np.add(weight_base[key], weight_to_add[key])
-            weight_base[key] = weight_base[key].tolist()
         return weight_base
 
     def accept(self, shareable: Shareable, fl_ctx: FLContext) -> bool:
-        result = shareable.get(MsgKey.RESULT)
-        weight_to_add = result.get(MsgKey.WEIGHTS)
-        mode = result.get(MsgKey.MODE)
-        if weight_to_add is None:
-            return True
-        if mode != self.mode:
-            return True
-
-        # bottom level does not have num_devices
-        # in which case num_devices_to_add is 1
-        num_devices_to_add = shareable.get(MsgKey.NUM_DEVICES)
-        if num_devices_to_add is None:
-            num_devices_to_add = 1
-        self.num_devices += num_devices_to_add
-        self.log_info(fl_ctx, f"Accepting result with {num_devices_to_add} devices")
+        dxo = from_shareable(shareable)
+        # check data_kind
+        if dxo.data_kind != DataKind.WEIGHT_DIFF:
+            raise ValueError(f"DXO data_kind must be {DataKind.WEIGHT_DIFF}, but got {dxo.data_kind}")
 
-        # add new weights to the existing weights
-        if self.weights is None:
-            self.weights = weight_to_add
+        # get weights and add to base
+        weight_to_add = dxo.data.get("dict")
```

Comment:
this "dict" is a magic key better be exposed to the users
either a constant or configurable argument in `__init__`?